### DISCUSSION TITLE
Scope Clarity stylesheet

### DISCUSSION THREAD

**Discussion by mbritense** [vmware-clarity/ng-clarity#325] [Q&A] [answered]
The product my company develops currently uses Bootstrap (and a theme). We are very interested in gradually refactoring our product to use Clarity. Our product is built in Angular. For testing purposes, I followed the instructions 'Adding Clarity to an Angular project' [here](https://clarity.design/documentation/get-started). Using Clarity components works. However, the stylesheet `node_modules/@clr/ui/clr-ui.min.css` seems to override a lot of the Bootstrap styling we use and breaks a lot of the application. Is there a way to (temporarily) scope the Clarity stylesheet to certain components until we refactor the entire application?

Thank you!

Two screenshots to illustrate the problem. I would only like the list on the screen to use the Clarity datagrid and the rest of the application to stay the same. (Never mind the data grid not displaying all properties correctly, this was just a quick test.)

Before:

![image](https://user-images.githubusercontent.com/79840403/193325077-efbd80d2-d2c9-4ab1-b0e2-0524d8bed2e9.png)

After:
![image](https://user-images.githubusercontent.com/79840403/193324788-0ce66bcf-5db8-40b7-b349-88ba96eb2b8b.png)

**Comment by ashleyryan** [comment, marked as answer]
clr/ui (the html/CSS implementation that clr/angular is built on) grew out of bootstrap years ago, so I'm not surprised to hear it's clobbering your bootstrap css. Unfortunately, it's very global at the moment and fixing that would be a massive undertaking. 

Clarity Core is built using web components (we also provide angular wrappers) and the CSS for those are scoped the component itself, no possibility of clobbering your CSS. https://core.clarity.design/ There's not as many components as in clr/angular though.

**Reply by mbritense** [reply]
> clr/ui (the html/CSS implementation that clr/angular is built on) grew out of bootstrap years ago, so I'm not surprised to hear it's clobbering your bootstrap css. Unfortunately, it's very global at the moment and fixing that would be a massive undertaking.
> 
> Clarity Core is built using web components (we also provide angular wrappers) and the CSS for those are scoped the component itself, no possibility of clobbering your CSS. https://core.clarity.design/ There's not as many components as in clr/angular though.

Thank you for your swift and clear response! Is there a roadmap available somewhere for when components are to be added to Clarity Core? And is it the aim to make all components available in Clarity Core eventually? Especially the [datagrid](https://clarity.design/documentation/datagrid/structure) is important for our purposes.
